plugins/social_network/social_network.cpp

```diff
@@ -790,15 +802,38 @@ namespace golos { namespace plugins { namespace social_network {
         auto query = args.args->at(0).as<discussion_query>();
         query.prepare();
         query.validate();
+        FC_ASSERT(!!query.start_author, "Must get comments for a specific author");
 
         auto& db = pimpl->database();
         return db.with_weak_read_lock([&]() {
-            return pimpl->select_ordered_discussions<sort::by_updated>(
-                query,
-                [&](const discussion& d) -> bool {
-                    return true;
+            const auto &idx = db.get_index<comment_index>().indices().get<by_author_last_update>();
+            auto itr = idx.lower_bound(*query.start_author);
+            FC_ASSERT(itr != idx.end(), "Comment is not in account's comments");
```

Comment:
Assert message looks wrong, here is author check, not comment.

Reply:
Fixed.